src/components/article-body/body-components/codeblock.tsx

```diff
@@ -10,15 +12,32 @@ export const CodeBlock = ({ lang, value }: { lang: string; value: string }) => {
     const numberOfLines = getNumberOfLines(value);
     const height = getCodeHeight(numberOfLines);
     const language = getFloraLanguage(lang);
+
+    // This is a workaround to a bug where a long line of code will be cut off at the end.
+    // This seems to be a CodeMirror issue, but is especially apparent here without line wrapping.
+    // Basically this will force the component to rerender after 1 second, which allows it to recalculate the line length.
+    // https://github.com/codemirror/codemirror5/issues/5639 describes a similar issue.
+
+    const [, setReady] = useState(false);
+    useEffect(() => {
+        setTimeout(() => {
+            setReady(true);
+        }, 1000);
+    }, []);
```

Comment:
I'm open to suggestions if someone knows a better way to do this.

Reply:
https://reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdate

if there's no way to fix this at the Code Mirror level, this forcing of an update might be a _little_ cleaner so we can avoid using `setTimeout` and you can just call the `forceUpdate()` in the hook

Reply:
seems like this approach doesn't work when the user has a slow internet connection. tested it with "Fast 3G" network throttling and the code is cut off at the end again for me.

not sure exactly what the issue is with codemirror, but is it possible to use either a [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) or a [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) to trigger a re-render once whatever change which is causing the problem (either visual or dom tree) is complete?

Reply:
@max-melo Ah good catch on that, yeah I'm not exactly sure what the Codemirror issue is, but I will look  into those APIs to see if I can determine when the code block has fully rendered and then try to get it to recalculate. 

@justinmckenzie yeah I agree I think that would be a little cleaner, I think I'll try to combine this with Max's suggestion.